### PR TITLE
list_slice alias for array_slice

### PR DESCRIPTION
**Pull request by makalaaneesh**
Hi, 

This addresses https://github.com/duckdb/duckdb/issues/2795

### CODE REVIEW

File: src/function/scalar/list/array_slice.cpp [hunk 1]
```diff
@@ -193,6 +193,8 @@ void ArraySliceFun::RegisterFunction(BuiltinFunctions &set) {
 	                   ArraySliceFunction, false, ArraySliceBind);
 	fun.varargs = LogicalType::ANY;
 	set.AddFunction(fun);
+	fun.name = "list_slice"; // alias to conform to list_* name scheme
```
Comment (lnkuiper) [inline src/function/scalar/list/array_slice.cpp:196]: We have a unified way to add functions with multiple aliases, see for instance `list_concat.cpp`.
Could you add the function with multiple aliases in the same way?

i.e. we create the function without name, then add it with an array of names:
```c++
ScalarFunction fun({LogicalType::ANY, LogicalType::BIGINT, LogicalType::BIGINT}, LogicalType::ANY, ArraySliceFunction, false, ArraySliceBind);
set.AddFunction({"array_slice", "list_slice"});
```

**Review by lnkuiper — CHANGES_REQUESTED** [review]
Thanks! I have a minor nitpick, then it can be merged.

File: src/function/scalar/list/array_slice.cpp (hunk 1 above)
Comment (makalaaneesh) [reply src/function/scalar/list/array_slice.cpp:196]: Ah, that's certainly better. I'll make the change 👍 
I saw this approach being used in `base64.cpp`, so I did the same. I'll go ahead and make the change there as well!

**Review by lnkuiper — APPROVED** [review]
Looks good now, thanks!

**Pull request merged**